src/internal/connector/exchange/event_calendar_cache.go

```diff
@@ -44,7 +44,9 @@ func (ecc *eventCalendarCache) populateEventRoot(ctx context.Context) error {
 		return errors.Wrap(err, "fetching calendar "+support.ConnectorStackErrorTrace(err))
 	}
 
-	temp := graph.NewCacheFolder(f, path.Builder{}.Append(container))
+	temp := graph.NewCacheFolder(f,
+		path.Builder{}.Append(container), // storage path
```

Comment:
this should be the ID
```suggestion
		path.Builder{}.Append(*f.GetId()), // storage path
```

Reply:
next PR!